### PR TITLE
TC-ACE-1.2

### PR DESCRIPTION
**Pull request by cecille**
automation for TC-ACE-1.2

AutoResubscribe:
If we error on initial subscription because we're NEVER going to
have access, the python scripts will never return - they will
continue to resubscribe forever and the future never completes.

This flag prevents any resubscription attempts. It's used in this
test because I know we're going to get a failure and I'd rather
have the function return quickly with the required information.

### CODE REVIEW

File: src/python_testing/TC_ACE_1_2.py [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+
+import logging
+import queue
+
+import chip.clusters as Clusters
+from chip.clusters import ClusterObjects as ClustersObjects
```
Comment (mlepage-google) [inline src/python_testing/TC_ACE_1_2.py:22]: Is this subtle rename intended or a typo?

File: src/python_testing/TC_ACE_1_2.py [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+        self.breadcrumb = self.breadcrumb + 1
+
+    async def steps_admin_subscription_error(self, print_steps: bool):
+        if print_steps:
+            self.print_step(13, "Subscribe to the ACL attribute, expect INVALID_ACTION")
+        try:
+            await self.TH2.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, Clusters.AccessControl.Attributes.Acl)], reportInterval=(1, 5), fabricFiltered=False, keepSubscriptions=False, autoResubscribe=False)
+            asserts.fail("Incorrectly subscribed to attribute with invalid permissions")
+        except ChipStackError as e:
+            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subcription with no permission")
```
Comment (mlepage-google) [inline src/python_testing/TC_ACE_1_2.py:114]:
```suggestion
            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subscription with no permission")
```

File: src/python_testing/TC_ACE_1_2.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+
+        if print_steps:
+            self.print_step(14, "Subscribe to the AccessControlEntryChanged event, expect INVALID_ACTION")
+        try:
+            await self.TH2.ReadEvent(nodeid=self.dut_node_id, events=[(0, Clusters.AccessControl.Events.AccessControlEntryChanged)], reportInterval=(
+                1, 5), fabricFiltered=False, keepSubscriptions=False, autoResubscribe=False)
+            asserts.fail("Incorrectly subscribed to event with invalid permissions")
+        except ChipStackError as e:
+            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subcription with no permission")
```
Comment (mlepage-google) [inline src/python_testing/TC_ACE_1_2.py:123]:
```suggestion
            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subscription with no permission")
```

File: src/python_testing/TC_ACE_1_2.py [hunk 4]
```diff
@@ -0,0 +1,272 @@
...
+
+        self.print_step(28, "TH2 repeats subscriptions to Admin attribute and event to ensure it still errors")
+        await self.steps_admin_subscription_error(print_steps=False)
+
+        self.print_step(29, "TH2 attempts to subscribe to the breadcrumb attribute - expect error")
+        try:
+            await self.TH2.ReadAttribute(nodeid=self.dut_node_id, attributes=[(0, Clusters.GeneralCommissioning.Attributes.Breadcrumb)], reportInterval=(1, 5), keepSubscriptions=False, autoResubscribe=False)
+            asserts.fail("Incorrectly subscribed to attribute with invalid permissions")
+        except ChipStackError as e:
+            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subcription with no permission")
```
Comment (mlepage-google) [inline src/python_testing/TC_ACE_1_2.py:268]:
```suggestion
            asserts.assert_equal(e.err, 0x580, "Incorrect error message received from subscription with no permission")
```

**Review by mlepage-google — APPROVED** [review]

File: src/python_testing/TC_ACE_1_2.py (hunk 1 above)
Comment (cecille) [reply src/python_testing/TC_ACE_1_2.py]: ha, no - not intended. Copied it from a another script with this error.

**Review by andy31415 — APPROVED** [review]

**Review by jmartinez-silabs — APPROVED** [review]

**Pull request merged**